src/Engine/IO/Window.cpp

```diff
@@ -118,18 +157,28 @@ void Window::update(float dt) {
 
 #ifdef DEBUG_MODE
     /* Update ImGui */
+    // TODO: clean up this code
+    static bool priorCursorState;
     imGuiTimer += dt;
-    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && 
-       (Keyboard::isKeyPressed(GLFW_KEY_LEFT_SHIFT) || Keyboard::isKeyPressed(GLFW_KEY_RIGHT_SHIFT)) &&
-        imGuiTimer >= 0.5) {
+    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && imGuiTimer >= 0.5) {
+        if (!isImGuiEnabled()) {
+            priorCursorState = cursorEnabled;
+            setCursorEnabled(true);
+        }
+
         toggleImGui();
         imGuiTimer = 0.0;
+        ImGui_ImplGlfwGL3_NewFrame(isImGuiEnabled());
```

Comment:
As I understand it, the boolean within NewFrame should be true. Is that not the case?

Reply:
fixed